src/metabase/api/alert.clj

```diff
@@ -20,12 +20,20 @@
 
 (api/defendpoint GET "/"
   "Fetch all alerts"
-  [archived]
-  {archived (s/maybe su/BooleanString)}
-  (as-> (pulse/retrieve-alerts {:archived? (Boolean/parseBoolean archived)}) <>
+  [archived user_id]
+  {archived (s/maybe su/BooleanString)
+   user_id  (s/maybe su/IntGreaterThanZero)}
+  (as-> (pulse/retrieve-alerts {:archived? (Boolean/parseBoolean archived)
+                                :user-id   user_id}) <>
     (filter mi/can-read? <>)
     (hydrate <> :can_write)))
 
+(api/defendpoint GET "/:id"
+  "Fetch an alert by ID"
+  [id]
+  (-> (api/read-check (pulse/retrieve-notification id))
```

Comment:
It seems like this could be used to fetch any sort of Pulse (not just Alerts) -- do we want to limit the endpoint to just Alerts? `GET /` is using `retrieve-alerts` which IIRC filters out things that aren't Alerts... I don't remember if there's a plain `retrieve-alert` we can use here

Reply:
That's fair, I can add a `retrieve-alert`